### PR TITLE
configcore: ignore system.pi-config.* setting on measured kernels

### PR DESCRIPTION
**Pull request by mvo5**
Some pi devices uses measured boot, e.g. avnet. On these devices
the config.txt is part of the measurements and we cannot modify
config.txt without changing the measurements and breaking the
 boot.
    
So this commit adds a check for kernels that known to use measured
boot and here the configuration is ignored.

Build on top of https://github.com/snapcore/snapd/pull/10446

### CODE REVIEW

File: overlord/configstate/configcore/picfg.go [hunk 1]
```diff
@@ -111,21 +133,21 @@ func piConfigFile(dev sysconfig.Device, opts *fsOnlyContext) (string, error) {
 		} else {
 			// we don't support configuring pi-config in these modes as it is
 			// unclear what the right behavior is
-			return "", errPiConfigNotSupported
+			return "", newPiConfigNotSupportedError("unsupported mode")
```
Comment (pedronis) [inline overlord/configstate/configcore/picfg.go:136]: "unsupported system mode"

File: overlord/configstate/configcore/picfg.go [hunk 2]
```diff
@@ -92,13 +93,34 @@ func updatePiConfig(path string, config map[string]string) error {
...
+	// see https://bugs.launchpad.net/denver/+bug/1928613
+	"avnet-avt-iiotg20-kernel",
+}
 
 func piConfigFile(dev sysconfig.Device, opts *fsOnlyContext) (string, error) {
 	rootDir := dirs.GlobalRootDir
 	subdir := "/boot/uboot"
+
+	if strutil.ListContains(piMeasuredBootKernels, dev.Kernel()) {
+		return "", newPiConfigNotSupportedError("kernel measures config.txt")
```
Comment (pedronis) [inline overlord/configstate/configcore/picfg.go:121]: s/kernel/boot/ perhaps ?

File: overlord/configstate/configcore/picfg_test.go [hunk 3]
```diff
@@ -251,3 +252,55 @@ func (s *piCfgSuite) TestFilesystemOnlyApply(c *C) {
...
+func (s *piCfgSuite) TestConfigurePiConfigSkippedOnWrongMode(c *C) {
+	logbuf, r := logger.MockLogger()
+	defer r()
+
+	if _, isSet := os.LookupEnv("SNAPD_DEBUG"); !isSet {
+		os.Setenv("SNAPD_DEBUG", "1")
+		defer os.Unsetenv("SNAPD_DEBUG")
+	}
+
+	coreDevInstalling := mockDev{
```
Comment (pedronis) [inline overlord/configstate/configcore/picfg_test.go:289]: maybe for consistency:

uc20DevInstallMode

**Review by pedronis — APPROVED** [review]
thank you, small comments

**Review by bboozzoo — APPROVED** [review]
LGTM

**Review by mardy — APPROVED** [review]

**Pull request merged**